### PR TITLE
Added support for a JSON health endpoint

### PR DESCRIPTION
**Pull request by aleksihakli**
Rebased #31 on top of the current master to get it finally merged.

### CODE REVIEW

File: health_check/urls.py [hunk 1]
```diff
@@ -2,10 +2,11 @@
...
 
 import health_check
-from health_check.views import home
+from health_check.views import home, json_view
 
 health_check.autodiscover()
 
 urlpatterns = [
     url(r'^$', home, name='health_check_home'),
+    url(r'^json/?$', json_view, name='health_check_json_view')
```
Comment (codingjoe) [inline health_check/urls.py]: What do you think of not add a separate path but making the view header sensitive? You could request the response to be a json.
```
Accept: application/json
```

File: health_check/backends/base.py [hunk 2]
```diff
@@ -15,6 +15,12 @@ class HealthCheckStatusType(object):
     2: _("unexpected result"),
 }
 
+HEALTH_CHECK_JSON_STATUS_TYPE_TRANSLATOR = {
```
Comment (codingjoe) [inline health_check/backends/base.py:18]: This could be an enum.

File: tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from django.test import TestCase
+
+from health_check.plugins import plugin_dir
+from health_check_cache.plugin_health_check import CacheBackend
+
+try:
+    # Django >= 1.4, < 2.0 import
+    from django.core.urlresolvers import reverse
```
Comment (codingjoe) [inline tests/test_views.py]: No need to add support for any version prior to 1.8

**Review by codingjoe — CHANGES_REQUESTED** [review]
In general I like the idea.
I don't know if it's not better to use literals instead of integers in the JSON.

Besides that, you really should add some documentation here.

File: health_check/urls.py (hunk 1 above)
Comment (aleksihakli) [reply health_check/urls.py]: That's a good idea. I'll just add a different view per Accept header and add some documentation :)

File: health_check/backends/base.py (hunk 2 above)
Comment (aleksihakli) [reply health_check/backends/base.py:18]: Should we refactor the original `HEALTH_CHECK_STATUS_TYPE_TRANSLATOR` as well, then? The current implementation uses the same logic for both; look up pretty response status based on the status code returned by the backend.

**Comment by aleksihakli** [comment]
@codingjoe I refactored the view to return the status with the `Accept: application/json` header and added tests for the JSON output. I'll rebase the commit chain on top of the 2.0 rewrite later to keep the old attributions here.

**Comment by aleksihakli** [comment]
Up-to-date PR in #80.

**Pull request closed without merging**